Classes/git/PBGitRevList.mm

```diff
@@ -129,12 +129,6 @@ - (void) addGitObject:(GTObject *)obj toCommitSet:(NSMutableSet *)set
 
 	NSAssert(commit, @"Can't add nil commit to set");
 
-	for (GTCommit *item in set) {
-		if ([item.OID isEqual:commit.OID]) {
-			return;
-		}
-	}
-
 	[set addObject:commit];
```

Comment:
`GTObject`s [implement `-hash` and `-isEqual` using the object hash][0], so they do indeed work correctly in sets.

[0]: https://github.com/libgit2/objective-git/blob/968396ab1f398551c8f1c04b6e639c18713f20f4/ObjectiveGit/GTObject.m#L60-L68